### PR TITLE
support torchaudio rnnt loss as default

### PR DESCRIPTION
**Pull request by aheba**
Hello,

This PR add the support of the torchaudio RNN-T loss (compiled from warp-transducer).

We saw that the RNNT loss doesn't normalize over `input_seq`, so the loss is a little bit higher...

RNNT loss torchaudio:
```
epoch: 1, lr_adam: 3.00e-04, lr_wav2vec: 1.00e-04 - train loss: 1.40e+02 - valid loss: 1.16e+02, valid PER: 1.00e+02
epoch: 2, lr_adam: 3.00e-04, lr_wav2vec: 1.00e-04 - train loss: 95.39 - valid loss: 64.14, valid PER: 91.21
epoch: 3, lr_adam: 3.00e-04, lr_wav2vec: 1.00e-04 - train loss: 35.57 - valid loss: 17.67, valid PER: 22.56
epoch: 4, lr_adam: 3.00e-04, lr_wav2vec: 1.00e-04 - train loss: 19.28 - valid loss: 12.31, valid PER: 16.15
```
RNNT loss spbrain:
```
epoch: 1, lr_adam: 3.00e-04, lr_wav2vec: 1.00e-04 - train loss: 1.06 - valid loss: 7.76e-01, valid PER: 1.00e+02
epoch: 2, lr_adam: 3.00e-04, lr_wav2vec: 1.00e-04 - train loss: 6.28e-01 - valid loss: 2.57e-01, valid PER: 54.77
epoch: 3, lr_adam: 3.00e-04, lr_wav2vec: 1.00e-04 - train loss: 2.16e-01 - valid loss: 1.08e-01, valid PER: 23.30
epoch: 4, lr_adam: 3.00e-04, lr_wav2vec: 1.00e-04 - train loss: 1.27e-01 - valid loss: 8.16e-02, valid PER: 14.56
```

### CODE REVIEW

**Comment by aheba** [comment]
as discussed offline with @carolineechen,
Torchaudio use warp-RNNT https://github.com/1ytic/warp-rnnt for GPU, and warp-transducer for CPU https://github.com/HawkAaron/warp-transducer

the warp-RNNT must be normalized on the length of input sequence to have the right LOSS as discussed with @lorenlugosch,
https://github.com/1ytic/warp-rnnt/blob/2944982ba1a53f6d20ba74c14b72cdd32e335d56/core.cu#L133

let me open an issue in the warp-RNNT, if the issue is fixed, I assume that the next release will automatically be included in torchaudio

**Comment by mravanelli** [comment]
Thank you @aheba for this PR. It tried it on TIMIT and Librispeech and everything seems to work. Before merging it, I would suggest adding some comments in the readme file about the two version of RNNT now supported (i.e, torchadio and numba).

**Comment by aheba** [comment]
Let me add some comment on the support of both warp-rnnt from torchaudio and our implementation..

**Comment by aheba** [comment]
@mravanelli , I just notice when adding specific comment to the readme that if users are using torchaudio version < 0.10.0, they will not have the RNN-T Loss..
Let me highlight it in the readme and add some checking..

**Comment by aheba** [comment]
@mravanelli , LGTM for me,

Let open another PR for adding FastEmit regularisation to our Transducer Numba loss

**Comment by mravanelli** [comment]
Thank you @aheba! I just did some minor modifications in the README file. Everything is ready to be merged.

File: recipes/CommonVoice/ASR/transducer/hparams/train_fr.yaml [hunk 1]
```diff
@@ -181,6 +181,7 @@ log_softmax: !new:speechbrain.nnet.activations.Softmax
    apply_log: True
 
 transducer_cost: !name:speechbrain.nnet.losses.transducer_loss
+   use_torchaudio: True
```
Comment (csukuangfj) [inline recipes/CommonVoice/ASR/transducer/hparams/train_fr.yaml:184]: Is there a reason to select torchaudio's rnnt-loss by default?

**Comment by csukuangfj** [comment]
It's surprising that no one has noticed that torchaudio's rnnt_loss takes `logits` as input, i.e., the output of `nn.Linear`, not `log_probs`.
